op-service/sources/engine_client.go

```diff
@@ -113,13 +123,19 @@ func (s *EngineClient) NewPayload(ctx context.Context, payload *eth.ExecutionPay
 // There may be two types of error:
 // 1. `error` as eth.InputError: the payload ID may be unknown
 // 2. Other types of `error`: temporary RPC errors, like timeouts.
-func (s *EngineClient) GetPayload(ctx context.Context, payloadId eth.PayloadID) (*eth.ExecutionPayloadEnvelope, error) {
-	e := s.log.New("payload_id", payloadId)
+func (s *EngineClient) GetPayload(ctx context.Context, payloadInfo eth.PayloadInfo) (*eth.ExecutionPayloadEnvelope, error) {
+	e := s.log.New("payload_id", payloadInfo.ID)
 	e.Trace("getting payload")
 	var result eth.ExecutionPayloadEnvelope
-	err := s.client.CallContext(ctx, &result, "engine_getPayloadV3", payloadId)
+	var method string
+	if s.rollupCfg.IsEcotone(payloadInfo.Timestamp) {
```

Comment:
Upstream has an API for `func (b PayloadID) Is(versions ...PayloadVersion) bool` which seems a lot nicer to use than having to pass the timestamp around everywhere.

Reply:
The method `Is` is using`PayloadID` that is prefixed with payload version. But unfortunately, the prefixed `PayloadID` is only used by `Geth`, it's not an Ethereum specs. So other execution engine like op-erigon does not prefix the payload version to the payload ID, so we cannot use that in `op-node`.

Reply:
It seems not too hard to apply it to op-erigon, but have no idea about other clients. I agree that it may reduce complexity, but it means we should add it to OP stack exec engine spec. I'm not sure if it's a good idea because actually it's not required changes for OP stack. 🤔